RecoTracker/TkDetLayers/src/GeometricSearchTracker.cc

```diff
@@ -78,46 +78,100 @@ GeometricSearchTracker::~GeometricSearchTracker() {
 #include "DataFormats/SiPixelDetId/interface/PixelSubdetector.h"
 
 const DetLayer* GeometricSearchTracker::idToLayer(const DetId& id) const {
-  switch (id.subdetId()) {
-    case StripSubdetector::TIB:
-      return theTibLayers[theTrkTopo->tibLayer(id) - 1];
-      break;
-
-    case StripSubdetector::TOB:
-      return theTobLayers[theTrkTopo->tobLayer(id) - 1];
-      break;
-
-    case StripSubdetector::TID:
-      if (theTrkTopo->tidSide(id) == 1) {
-        return theNegTidLayers[theTrkTopo->tidWheel(id) - 1];
-      } else if (theTrkTopo->tidSide(id) == 2) {
-        return thePosTidLayers[theTrkTopo->tidWheel(id) - 1];
-      }
-      break;
-
-    case StripSubdetector::TEC:
-      if (theTrkTopo->tecSide(id) == 1) {
-        return theNegTecLayers[theTrkTopo->tecWheel(id) - 1];
-      } else if (theTrkTopo->tecSide(id) == 2) {
-        return thePosTecLayers[theTrkTopo->tecWheel(id) - 1];
-      }
-      break;
-
-    case PixelSubdetector::PixelBarrel:
-      return thePixelBarrelLayers[theTrkTopo->pxbLayer(id) - 1];
-      break;
-
-    case PixelSubdetector::PixelEndcap:
-      if (theTrkTopo->pxfSide(id) == 1) {
-        return theNegPixelForwardLayers[theTrkTopo->pxfDisk(id) - 1];
-      } else if (theTrkTopo->pxfSide(id) == 2) {
-        return thePosPixelForwardLayers[theTrkTopo->pxfDisk(id) - 1];
-      }
-      break;
-
-    default:
-      edm::LogError("TkDetLayers") << "ERROR:layer not found!";
-      // throw(something);
+  //If it's a tracker Det
+  if (id.det() == 1) {
```

Comment:
```suggestion
  if (id.det() == DetId::Detector::Tracker) {
```

Reply:
Hi Andrea,

Thanks for the suggestion. I have implemented it in a later commit of the code.